Paket.restore.targets creates nuspec in wrong folder

I just upgrade a solution to the new csproj format, but have a problem with packaging with `dotnet pack`
 afterwards. As I understand `paket.Restore.targets` override the packaging process by generating nuspec files which are modified afterward by `paket fix-nuspecs`. The problem is that these nuspec files are duplicated in `obj` and `obj/$(Configuration)`, and only last one is correct. There is one project which does not have a `paket.references` file and it is correctly generated.

As I work-a-round I tried deleting the entire section in the  `paket.Restore.targets` and now everything works as expected. Dependencies are also correct.

```
PS C:\Users\tda\tmp\paket-test> dotnet pack /p:Description=foobar
Microsoft (R) Build Engine version 15.4.8.50001 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  bar -> C:\Users\tda\tmp\paket-test\bar\bin\Debug\net461\bar.dll
  bar -> C:\Users\tda\tmp\paket-test\bar\bin\Debug\netstandard2.0\bar.dll
  foo -> C:\Users\tda\tmp\paket-test\foo\bin\Debug\net461\foo.dll
  Successfully created package 'C:\Users\tda\tmp\paket-test\bar\bin\Debug\bar.1.0.0.nupkg'.
  Paket version 5.119.8
  Performance:
   - Runtime: 1 second
  Paket failed with
  -> Could not parse nuspec file 'obj\foo.1.0.0.nuspec'.
  -> XmlException: Root element is missing.
C:\Users\tda\tmp\paket-test\.paket\Paket.Restore.targets(150,5): error MSB3073: The command ""C:\Users\tda\tmp\paket-test\.paket\paket.exe" fix-nuspecs files "obj\foo.1.0.0.nuspec" project
-file "C:\Users\tda\tmp\paket-test\foo/foo.csproj" " exited with code 1. [C:\Users\tda\tmp\paket-test\foo\foo.csproj]

any chance you could come up with a patch?

2017-10-26 14:47 GMT+02:00 Trond Danielsen <notifications@github.com>:

> PS C:\Users\tda\tmp\paket-test> dotnet pack /p:Description=foobar
> Microsoft (R) Build Engine version 15.4.8.50001 for .NET Core
> Copyright (C) Microsoft Corporation. All rights reserved.
>
>   bar -> C:\Users\tda\tmp\paket-test\bar\bin\Debug\net461\bar.dll
>   bar -> C:\Users\tda\tmp\paket-test\bar\bin\Debug\netstandard2.0\bar.dll
>   foo -> C:\Users\tda\tmp\paket-test\foo\bin\Debug\net461\foo.dll
>   Successfully created package 'C:\Users\tda\tmp\paket-test\bar\bin\Debug\bar.1.0.0.nupkg'.
>   Paket version 5.119.8
>   Performance:
>    - Runtime: 1 second
>   Paket failed with
>   -> Could not parse nuspec file 'obj\foo.1.0.0.nuspec'.
>   -> XmlException: Root element is missing.
> C:\Users\tda\tmp\paket-test\.paket\Paket.Restore.targets(150,5): error MSB3073: The command ""C:\Users\tda\tmp\paket-test\.paket\paket.exe" fix-nuspecs files "obj\foo.1.0.0.nuspec" project
> -file "C:\Users\tda\tmp\paket-test\foo/foo.csproj" " exited with code 1. [C:\Users\tda\tmp\paket-test\foo\foo.csproj]
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/fsprojects/Paket/issues/2865#issuecomment-339654275>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADgNOPqvonMr8bCylAok7KANccnQ-fcks5swH9FgaJpZM4QHfaK>
> .
>

--- a/.paket/Paket.Restore.targets
+++ b/.paket/Paket.Restore.targets
@@ -136,15 +136,21 @@
   </Target>
 
   <Target Name="PaketOverrideNuspec" AfterTargets="GenerateNuspec" Condition="('$(IsPackable)' == '' Or '$(IsPackable)' == 'true') And Exists('$(MSBuildProjectDirectory)/obj/$(MSBuildProjectFile).references')" >
+    <ItemGroup>
+      <_NuspecFilesNewLocation Include="$(BaseIntermediateOutputPath)$(Configuration)/*.nuspec"/>
+    </ItemGroup>
+
     <PropertyGroup>
       <PaketProjectFile>$(MSBuildProjectDirectory)/$(MSBuildProjectFile)</PaketProjectFile>
       <ContinuePackingAfterGeneratingNuspec>true</ContinuePackingAfterGeneratingNuspec>
       <UseNewPack>false</UseNewPack>
       <UseNewPack Condition=" '$(NuGetToolVersion)' != '4.0.0' ">true</UseNewPack>
+      <AdjustedNuspecOutputPath>$(BaseIntermediateOutputPath)$(Configuration)/</AdjustedNuspecOutputPath>
+      <AdjustedNuspecOutputPath Condition="@(_NuspecFilesNewLocation) == ''">$(BaseIntermediateOutputPath)</AdjustedNuspecOutputPath>
     </PropertyGroup>
 
     <ItemGroup>
-      <_NuspecFiles Include="$(BaseIntermediateOutputPath)*.nuspec"/>
+      <_NuspecFiles Include="$(AdjustedNuspecOutputPath)*.nuspec"/>
     </ItemGroup>
 
     <Exec Command='$(PaketCommand) fix-nuspecs files "@(_NuspecFiles)" project-file "$(PaketProjectFile)" ' Condition="@(_NuspecFiles) != ''" />
@@ -188,7 +194,7 @@
               Serviceable="$(Serviceable)"
               FrameworkAssemblyReferences="@(_FrameworkAssemblyReferences)"
               ContinuePackingAfterGeneratingNuspec="$(ContinuePackingAfterGeneratingNuspec)"
-              NuspecOutputPath="$(BaseIntermediateOutputPath)"
+              NuspecOutputPath="$(AdjustedNuspecOutputPath)"
               IncludeBuildOutput="$(IncludeBuildOutput)"
               BuildOutputFolder="$(BuildOutputTargetFolder)"
               ContentTargetFolders="$(ContentTargetFolders)"
@@ -230,7 +236,7 @@
               Serviceable="$(Serviceable)"
               AssemblyReferences="@(_References)"
               ContinuePackingAfterGeneratingNuspec="$(ContinuePackingAfterGeneratingNuspec)"
-              NuspecOutputPath="$(BaseIntermediateOutputPath)"
+              NuspecOutputPath="$(AdjustedNuspecOutputPath)"
               IncludeBuildOutput="$(IncludeBuildOutput)"
               BuildOutputFolder="$(BuildOutputTargetFolder)"
               ContentTargetFolders="$(ContentTargetFolders)"

--- a/src/Paket/embedded/Paket.Restore.targets
+++ b/src/Paket/embedded/Paket.Restore.targets
@@ -136,15 +136,21 @@
   </Target>
 
   <Target Name="PaketOverrideNuspec" AfterTargets="GenerateNuspec" Condition="('$(IsPackable)' == '' Or '$(IsPackable)' == 'true') And Exists('$(MSBuildProjectDirectory)/obj/$(MSBuildProjectFile).references')" >
+    <ItemGroup>
+      <_NuspecFilesNewLocation Include="$(BaseIntermediateOutputPath)$(Configuration)/*.nuspec"/>
+    </ItemGroup>
+
     <PropertyGroup>
       <PaketProjectFile>$(MSBuildProjectDirectory)/$(MSBuildProjectFile)</PaketProjectFile>
       <ContinuePackingAfterGeneratingNuspec>true</ContinuePackingAfterGeneratingNuspec>
       <UseNewPack>false</UseNewPack>
       <UseNewPack Condition=" '$(NuGetToolVersion)' != '4.0.0' ">true</UseNewPack>
+      <AdjustedNuspecOutputPath>$(BaseIntermediateOutputPath)$(Configuration)/</AdjustedNuspecOutputPath>
+      <AdjustedNuspecOutputPath Condition="@(_NuspecFilesNewLocation) == ''">$(BaseIntermediateOutputPath)</AdjustedNuspecOutputPath>
     </PropertyGroup>
 
     <ItemGroup>
-      <_NuspecFiles Include="$(BaseIntermediateOutputPath)*.nuspec"/>
+      <_NuspecFiles Include="$(AdjustedNuspecOutputPath)*.nuspec"/>
     </ItemGroup>
 
     <Exec Command='$(PaketCommand) fix-nuspecs files "@(_NuspecFiles)" project-file "$(PaketProjectFile)" ' Condition="@(_NuspecFiles) != ''" />
@@ -188,7 +194,7 @@
               Serviceable="$(Serviceable)"
               FrameworkAssemblyReferences="@(_FrameworkAssemblyReferences)"
               ContinuePackingAfterGeneratingNuspec="$(ContinuePackingAfterGeneratingNuspec)"
-              NuspecOutputPath="$(BaseIntermediateOutputPath)"
+              NuspecOutputPath="$(AdjustedNuspecOutputPath)"
               IncludeBuildOutput="$(IncludeBuildOutput)"
               BuildOutputFolder="$(BuildOutputTargetFolder)"
               ContentTargetFolders="$(ContentTargetFolders)"
@@ -230,7 +236,7 @@
               Serviceable="$(Serviceable)"
               AssemblyReferences="@(_References)"
               ContinuePackingAfterGeneratingNuspec="$(ContinuePackingAfterGeneratingNuspec)"
-              NuspecOutputPath="$(BaseIntermediateOutputPath)"
+              NuspecOutputPath="$(AdjustedNuspecOutputPath)"
               IncludeBuildOutput="$(IncludeBuildOutput)"
               BuildOutputFolder="$(BuildOutputTargetFolder)"
               ContentTargetFolders="$(ContentTargetFolders)"

